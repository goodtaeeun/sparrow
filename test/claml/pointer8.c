int doo(int x) { return x; }

int foo(int a, int (*decode)(int)) {
  int b = decode(a);
  return b;
}

int main() {
  int r = foo(3, doo);
  sparrow_print(r);
  int q = foo(3, &doo);
  return 0;
}
