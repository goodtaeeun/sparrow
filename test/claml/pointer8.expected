/* Generated by CIL v. 1.7.4 */
/* print_CIL_Input is false */

#line 1 "pointer8.c"
int doo(int x) {

  {
#line 1
    return (x);
  }
}
#line 3 "pointer8.c"
int foo(int a, int (*decode)(int)) {
  int b;
  int __cil_tmp4;

  {
    {
#line 4
      __cil_tmp4 = (*decode)(a);
#line 4
      b = __cil_tmp4;
    }
#line 5
    return (b);
  }
}
#line 8 "pointer8.c"
int main(void) {
  int r;
  int __cil_tmp2;
  int q;
  int __cil_tmp4;

  {
    {
#line 9
      __cil_tmp2 = foo(3, &doo);
#line 9
      r = __cil_tmp2;
#line 10
      sparrow_print(r);
#line 11
      __cil_tmp4 = foo(3, &doo);
#line 11
      q = __cil_tmp4;
    }
#line 12
    return (0);
  }
}
